homeassistant/components/nibe_heatpump/button.py

```diff
@@ -0,0 +1,55 @@
+"""The Nibe Heat Pump sensors."""
+from __future__ import annotations
+
+from homeassistant.components.button import ButtonEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import DOMAIN, Coordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up platform."""
+
+    coordinator: Coordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    async_add_entities([ResetButton(coordinator, "alarm-reset-45171", "alarm-45001")])
+
+
+class ResetButton(CoordinatorEntity[Coordinator], ButtonEntity):
+    """Sensor entity."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self, coordinator: Coordinator, reset_coil: str, alarm_coil: str
+    ) -> None:
+        """Initialize entity."""
+        self._reset_coil = coordinator.heatpump.get_coil_by_name(reset_coil)
+        self._alarm_coil = coordinator.heatpump.get_coil_by_name(alarm_coil)
+        super().__init__(coordinator, {self._alarm_coil.address})
+        self._attr_name = self._reset_coil.title
+        self._attr_unique_id = f"{coordinator.unique_id}-{self._reset_coil.name}"
+        self._attr_device_info = coordinator.device_info
+        self._attr_entity_category = EntityCategory.DIAGNOSTIC
```

Comment:
Same difference since there will only be one of these, but I could make this a class attribute.